llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -1198,6 +1198,18 @@ bool CombinerHelper::matchCombineExtractedVectorLoad(MachineInstr &MI,
   if (!VecEltTy.isByteSized())
     return false;
 
+  // Check for load fold barriers between the extraction and the load.
+  if (MI.getParent() != LoadMI->getParent())
+    return false;
+  const unsigned MaxIter = 20;
+  unsigned Iter = 0;
+  for (auto II = LoadMI->getIterator(), IE = MI.getIterator(); II != IE; ++II) {
```

Comment:
random aside: I didn't look too hard, but I'm surprised I couldn't find anything in STLExtras that automates this "do a think for a maximum number of iterations over a range" pattern

Like it would be cool to be able to write something like

```
if (any_of(make_bounded_range(LoadMI->getIterator(), MI.getIterator(), 20)
                              [const MachineInstr &Btwn]{
                                 return Btwn.isLoadFoldBarrier();
                              })
  return false;
```

But I don't know if that exists today.

Reply:
I'm not aware of an option.